base/java-tools/src/com/netscape/cmstools/HttpClient.java

```diff
@@ -152,7 +153,8 @@ public void send(String ifilename, String ofilename, String tokenName, String db
                     else
                         System.out.println("client cert is not null");
                     sslSocket.setUseClientMode(true);
-                    sslSocket.setClientCertNickname(nickname);
+                    String certnick = certname.toString();
```

Comment:
functionally looks good, but for cleanliness, see how "certname.toString()" is used at line 148 too?  How about move this (line 156) up above that (line 148) and change the reference of "certname.toString()"  to "certnick" as well?

Reply:
Thanks @ladycfu .
How about this:

                    sslSocket.setUseClientMode(true);
                    sslSocket.setClientCertNickname(certname.toString());
                    X509Certificate cert =
                        cm.findCertByNickname(certname.toString());

                    if (cert == null)
                        System.out.println("client cert is null");
                    else
                        System.out.println("client cert is not null");

Reply:
What I meant was:
                   **String certnick = certname.toString();**
                    X509Certificate cert =
                        cm.findCertByNickname(**certnick**);

                    if (cert == null)
                        System.out.println("client cert is null");
                    else
                        System.out.println("client cert is not null");
                    sslSocket.setUseClientMode(true);
                    sslSocket.setClientCertNickname(**certnick**);